orchagent/switchorch.cpp

```diff
@@ -900,6 +900,235 @@ void SwitchOrch::doCfgSwitchHashTableTask(Consumer &consumer)
+        {
+            if (!trimCap.validateQueueModeCap(trim.queue.mode.value))
+            {
+                SWSS_LOG_ERROR("Failed to validate switch trimming queue mode: capability is not supported");
+                return false;
+            }
+
+            if (!setSwitchTrimmingQueueModeSai(trim))
+            {
+                SWSS_LOG_ERROR("Failed to set switch trimming queue mode in SAI");
+                return false;
+            }
+
+            if (trimHlpr.isStaticQueueMode(tObj))
+            {
+                qIdxBak = true;
+            }
+
+            cfgUpd = true;
+        }
+    }
+    else
+    {
+        if (tObj.queue.mode.is_set)
+        {
+            SWSS_LOG_ERROR("Failed to remove switch trimming queue configuration: operation is not supported");
+            return false;
+        }
+    }
+
+    if (trim.queue.index.is_set)
+    {
+        if (!tObj.queue.index.is_set || (tObj.queue.index.value != trim.queue.index.value))
+        {
+            if (!setSwitchTrimmingQueueIndexSai(trim))
+            {
+                SWSS_LOG_ERROR("Failed to set switch trimming queue index in SAI");
+                return false;
+            }
+
+            cfgUpd = true;
+        }
+    }
+
+    // Don't update internal cache when config remains unchanged
+    if (!cfgUpd)
+    {
+        SWSS_LOG_NOTICE("Switch trimming in SAI is up-to-date");
+        return true;
+    }
+
+    if (qIdxBak) // Override queue index configuration during transition from static -> dynamic
+    {
+        auto cfg = trim;
+        cfg.queue.index = tObj.queue.index;
+        trimHlpr.setConfig(cfg);
+    }
+    else // Regular configuration update
+    {
+        trimHlpr.setConfig(trim);
+    }
+
+    SWSS_LOG_NOTICE("Set switch trimming in SAI");
+
+    return true;
+}
+
+void SwitchOrch::doCfgSwitchTrimmingTableTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto &map = consumer.m_toSync;
+    auto it = map.begin();
+
+    while (it != map.end())
+    {
+        auto keyOpFieldsValues = it->second;
+        auto key = kfvKey(keyOpFieldsValues);
+        auto op = kfvOp(keyOpFieldsValues);
+
+        SWSS_LOG_INFO("KEY: %s, OP: %s", key.c_str(), op.c_str());
+
+        if (key.empty())
+        {
+            SWSS_LOG_ERROR("Failed to parse switch trimming key: empty string");
+            it = map.erase(it);
+            continue;
+        }
+
+        SwitchTrimming trim;
+
+        if (op == SET_COMMAND)
+        {
+            for (const auto &cit : kfvFieldsValues(keyOpFieldsValues))
+            {
+                auto fieldName = fvField(cit);
+                auto fieldValue = fvValue(cit);
+
+                SWSS_LOG_INFO("FIELD: %s, VALUE: %s", fieldName.c_str(), fieldValue.c_str());
+
+                trim.fieldValueMap[fieldName] = fieldValue;
+            }
+
+            if (trimHlpr.parseConfig(trim))
+            {
+                if (!setSwitchTrimming(trim))
```

Comment:
Please check GCU update for all the new set of configs, so that these configs can be modified.

Reply:
@kperumalbfn / @r12f wip